rasa/nlu/model.py

```diff
@@ -307,25 +313,63 @@ def load(
             model_dir: The path of the model to load
             component_builder: The
                 :class:`rasa.nlu.components.ComponentBuilder` to use.
+            new_config: Optional new config to use for the new epochs.
+            finetuning_epoch_fraction: Value to multiply all epochs by.
 
         Returns:
             An interpreter that uses the loaded model.
         """
-
         model_metadata = Metadata.load(model_dir)
 
+        if new_config:
+            Interpreter._update_epochs_from_new_config(
+                model_metadata, new_config, finetuning_epoch_fraction
+            )
+
         Interpreter.ensure_model_compatibility(model_metadata)
-        return Interpreter.create(model_metadata, component_builder, skip_validation)
+        return Interpreter.create(
+            model_metadata,
+            component_builder,
+            skip_validation,
+            should_finetune=new_config is not None,
+        )
+
+    @staticmethod
+    def _update_epochs_from_new_config(
+        model_metadata: Metadata,
+        new_config: Optional[Dict] = None,
+        finetuning_epoch_fraction: float = 1.0,
+    ):
+        for p1, p2 in zip(model_metadata.metadata["pipeline"], new_config["pipeline"]):
+            if not p1.get("name") == p2.get("name"):
```

Comment:
Just checking `name` is not enough. Other parameters should also match exactly(except `epochs`).

Reply:
Also would be useful if we can mention in the raised exception which parameter does not match.

Reply:
Looking at other [PR](https://github.com/RasaHQ/rasa/pull/7427/files), it seems like you already check it there. Do we still need the check here then?

Reply:
So you are right that if anything apart from the epochs change we will not be able to fine-tune (due to the fingerprint checks). I included this name check as a sanity check - but maybe it's unnecessary.

Reply:
Yeah, I think they are a bit redundant. The fingerprint checks preserve the order of components in the pipeline as well right?

Reply:
No strong feelings about this duplicate check.

Reply:
ok then will remove.